common/Freemium.cpp

```diff
@@ -0,0 +1,44 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+#include <string>
+#include <vector>
+#include "ConfigUtil.hpp"
+#include "Util.hpp"
+#include "Freemium.hpp"
+
+#include <boost/algorithm/string/trim.hpp>
+
+namespace Freemium {
+    std::vector<std::string> FreemiumManager::FreemiumBlacklist;
+
+    FreemiumManager::FreemiumManager() {}
+
+    const std::vector<std::string>& FreemiumManager::getFreemiumBlacklist()
+    {
+#ifdef ENABLE_FREEMIUM
+        // Return if array is already filled
+        if (FreemiumBlacklist.size() != 0)
+            return FreemiumBlacklist;
+
+        StringVector commandList;
+        if (config::getString("freemium.disabled_commands", "") != "")
+            commandList = Util::tokenize(config::getString("freemium.disabled_commands", ""), "\n");
+
+        std::string command;
```

Comment:
Unused variable? 

Please enable all warning (`-Wall`) in your compiler. You can do this by setting CXXFLAGS="-Wall" when running configure. Something like `./configure CXXFLAGS="-Wall" <other flags>`